apps/ogr2ogr_lib.cpp

```diff
@@ -5271,51 +5275,14 @@ int LayerTranslator::Translate(OGRFeature *poFeatureIn, TargetLayerInfo *psInfo,
                 if (nDstGeomFieldCount == 0 && poStolenGeometry &&
                     m_poClipSrcOri)
                 {
-                    auto poGeomSRS = poStolenGeometry->getSpatialReference();
-                    if (m_poClipSrcReprojectedToSrcSRS_SRS != poGeomSRS)
-                    {
-                        auto poClipSrcSRS =
-                            m_poClipSrcOri->getSpatialReference();
-                        if (poClipSrcSRS && poGeomSRS &&
-                            !poClipSrcSRS->IsSame(poGeomSRS))
-                        {
-                            // Transform clip geom to geometry SRS
-                            m_poClipSrcReprojectedToSrcSRS.reset(
-                                m_poClipSrcOri->clone());
-                            if (m_poClipSrcReprojectedToSrcSRS->transformTo(
-                                    poGeomSRS) != OGRERR_NONE)
-                            {
-                                delete poStolenGeometry;
-                                goto end_loop;
-                            }
-                            m_poClipSrcReprojectedToSrcSRS_SRS = poGeomSRS;
-                        }
-                        else if (!poClipSrcSRS && poGeomSRS)
-                        {
-                            if (!m_bWarnedClipSrcSRS)
-                            {
-                                m_bWarnedClipSrcSRS = true;
-                                CPLError(
-                                    CE_Warning, CPLE_AppDefined,
-                                    "Clip source geometry has no attached SRS, "
-                                    "but the feature's geometry has one. "
-                                    "Assuming clip source geometry SRS is the "
-                                    "same as the feature's geometry");
-                            }
-                        }
-                    }
-                    OGRGeometry *poClipped = poStolenGeometry->Intersection(
-                        m_poClipSrcReprojectedToSrcSRS
-                            ? m_poClipSrcReprojectedToSrcSRS.get()
-                            : m_poClipSrcOri);
-                    delete poStolenGeometry;
-                    poStolenGeometry = nullptr;
-                    if (poClipped == nullptr || poClipped->IsEmpty())
+                    const OGRGeometry *poClipGeom =
+                        GetSrcClipGeom(poStolenGeometry->getSpatialReference());
+
+                    if (poClipGeom != nullptr &&
+                        !poClipGeom->Intersects(poStolenGeometry))
```

Comment:
the call to Intersection() is missing

Reply:
If I understood right, it's not needed here -- this path is for the case when we're writing to a dataset with no geometry, so the clip operation is just being used for filtering.

Reply:
ah, you're right